src/Adapters/DatabaseApproverRepository.php

```diff
@@ -75,6 +75,25 @@ public function setApproverCategories( int $userId, array $categoryNames ): void
 		);
 	}
 
+	/**
+	 * @return string[]
+	 */
+	public function getAllCategories(): array {
+		$result = $this->database->select(
+			'approver_config',
+			[ 'ac_categories AS categories' ],
+			[],
+			__METHOD__
+		);
+
+		$allCategories = [];
+		foreach ( $result as $row ) {
+			$allCategories = array_merge( $allCategories, $this->deserializeCategories( (string)$row->categories ) );
+		}
+
+		return array_unique( $allCategories );
```

Comment:
What about the one-liner I posted :(

Reply:
Sorry! I couldn't get back early. I will create a small PR for refactoring.